### PR TITLE
feat!: drop Laminas v2 support

### PR DESCRIPTION
**Pull request by JoshuaLicense**
## Description
Drop support for Laminas v2.
Add support for Laminas v3.

### CODE REVIEW

File: src/Command/Application/CreateSnapshot.php [hunk 1]
```diff
@@ -17,12 +18,6 @@
  */
 final class CreateSnapshot extends AbstractCommand
 {
-    const ON_SUBMIT = 0;
```
Comment (ilindsay) [inline src/Command/Application/CreateSnapshot.php:20]: This isn't great...but these constants are accessed outside of the library - backend definitely uses them. So you'd need to replicate them there if removing them here.

Comment (JoshuaLicense) [reply src/Command/Application/CreateSnapshot.php:20]: Okay, I've put them back and turned all the visibility back to `public`.

**Review by fibble — APPROVED** [review]
Skimmed through few 100 first file. Approach is sound, haven't fully reviewed all code - best tested by using it i think.

File: test/Util/DtoTest.php [hunk 2]
```diff
@@ -21,16 +20,15 @@
...
     /**
      * @var AnnotationBuilder
      */
     protected $annotationBuilder;
 
     public function setUp(): void
     {
-        $annotationBuilder = new AnnotationBuilder();;
+        $annotationBuilder = new AnnotationBuilder();
+        ;
```
Comment (hobbyhacker0) [inline test/Util/DtoTest.php:31]: Odd cleanup

Comment (JoshuaLicense) [reply test/Util/DtoTest.php:31]: Auto-fix by the static analysis tool

Comment (JoshuaLicense) [reply test/Util/DtoTest.php]: Removed it.

**Review by fibble — APPROVED** [review]
Skimmed through few 100 first file. Approach is sound, haven't fully reviewed all code - best tested by using it i think.

**Review by hobbyhacker0 — APPROVED** [review]

**Pull request merged**